Userland/Libraries/LibAudio/WavWriter.cpp

```diff
@@ -27,8 +27,7 @@ WavWriter::WavWriter(int sample_rate, u16 num_channels, PcmSampleFormat sample_f
 
 WavWriter::~WavWriter()
 {
-    if (!m_finalized)
-        finalize();
+    VERIFY(m_finalized);
 }
```

Comment:
I don't understand why your mandatory finalize method is better than this pattern that you removed?

It's akin to creating a File object that requires you flush() the file to disk and abort()s if you don't. No File-like API I know of has that behavior. We have destructors for a reason, to dispose of objects properly. 

If you wanted to handle any possible errors from flushing the file to disk, you would have called fflush() yourself before closing it. Most applications don't actually care that much, so we shouldn't have the finalize API be mandatory.

Reply:
We discussed that this new method is supposed to be better, since finalization should be mandatory as it doesn’t just leave files open, it leaves them in an inconsistent state. Non-finalized WAVs are unreadable by most tools.

Reply:
...sure, but we could just have the base class finalize if not finalized? Having a virtual function that's mandatory seems odd is my point, when we have a destructor to ensure a consistent state on destruction.

Reply:
or what about this: 

I call finalize on an Encoder object, but it fails, and doesn't set the m_finalized flag. 

Now what? Can I call finalize again? What if that fails? Can I destroy the object? If I delete it, will it crash my program?

An object in this now failed-to-finalize state is now just a zombie that I have to carry around, not knowing what operation are valid on it.

I would much rather the finalize API not be a "mandatory or we crash your program" deal. If you forgot to finalize, it will be finalized at destruction seems much more reasonable. If you finalize early, you can be sure that your encoded data has been sync'd to disk and final checksums written, etc.

Reply:
I do not have strong opinions either way, but I'd like to not go back and forth on this constantly.